src/Bufferer/PipeBufferer.php

```diff
@@ -14,38 +13,43 @@ class PipeBufferer implements BuffererInterface
 {
     private $logger;
     private $inputStream;
-    private $outputStream;
 
     private $mimeType;
     private $filePath;
     private $progressBar;
 
     private $buffering = true;
 
-    /**
-     * @param resource $inputStream
-     */
+    private $deferred;
+
     public function __construct(
         LoggerInterface $logger,
-        $inputStream,
+        InputStream $inputStream,
         ?string $outputPath,
         ConsoleSectionOutput $output
     ) {
         $this->logger = $logger;
-        $this->inputStream = new ResourceInputStream($inputStream);
-        $this->outputStream = new ResourceOutputStream($fileOutput = $outputPath ? fopen($outputPath, 'w') : tmpfile());
+        $this->inputStream = $inputStream;
         $this->mimeType = new \Amp\Deferred;
-        $this->filePath = $outputPath ?: stream_get_meta_data($fileOutput)['uri'];
+        $this->filePath = $outputPath ?? \tempnam(\sys_get_temp_dir(), 'stdinho') . '.tmp';
```

Comment:
`?:` checks for truthiness too, so it would treat `'0'` as false (though here it's an unlikely filename). Truthiness is always issue to be aware of with strings. I like `??` because it only checks to see the variable is not null, and of course with a function parameter it always will be set.

Reply:
Regarding using the temporary file, perhaps it could be unlinked as part of the destructor or `tmpfile()` could be used again and reopened using the meta data url.

Reply:
Yep, I am well aware of this

Reply:
There's no need for it to be static… just a clean up method on `Command`.

Reply:
The manual says: `If the script terminates unexpectedly, the temporary file may not be deleted.` Does that include signals, so is `tmpfile()` affected as well?

If you use a signal handler, that simply calls `exit`, the destructors will be run.

Reply:
Indeed. Still, this is workaround for usage of amphp/file which I think won't be used in the end